HoRNDIS.cpp

```diff
@@ -582,182 +417,297 @@ bool HoRNDIS::createNetworkInterface() {
 }
 
 /***** Interface enable and disable logic *****/
+HoRNDIS::ReentryLocker::ReentryLocker(IOCommandGate *inGate, bool &inGuard)
+		: gate(inGate), entryGuard(inGuard), result(kIOReturnSuccess) {
+	// Wait until we exit the previously-entered enable or disable method:
+	while (entryGuard) {
+		LOG(V_DEBUG, "Delaying the re-entered call");
+		result = gate->commandSleep(&entryGuard);
+		// If "commandSleep" has failed, stop immediately, and don't
+		// touch the 'entryGuard':
+		if (isInterrupted()) {
+			return;
+		}
+	}
```

Comment:
Hmm, ok.  So here's the thing that I still don't quite understand.  When can this reentrancy occur?  If it's only on a single thread, then blocking the reentrancy should result in a deadlock; if it's on multiple threads, then there's a race condition between the exit of the `while` loop (line 423) and the assertion of `entryGuard` (line 432), right?

I think that, at the very least, it would probably be beneficial to have a comment explaining why lines 423 and 432 do not race ... that pattern is somewhat concerning at first look.

Reply:
I added a reference back to the header file comment. Also, please see my other email dedicated to that subject.